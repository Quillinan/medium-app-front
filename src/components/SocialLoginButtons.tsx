const SocialLoginButtons: React.FC = () => {
  return (
    <div className='flex flex-col space-y-4'>
      <button
        className='flex w-full justify-center rounded-md bg-blue-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-blue-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600'
        onClick={() => alert('Continue with Azure')}
      >
        Login with Azure
      </button>
      <button
        className='flex w-full justify-center rounded-md bg-gray-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-gray-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={() => alert('Continue with GitHub')}
      >
        Login with GitHub
      </button>
    </div>
  );
};

export default SocialLoginButtons;
