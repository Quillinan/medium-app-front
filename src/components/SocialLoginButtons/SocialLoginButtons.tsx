import { useMsal } from '@azure/msal-react';
import microsoft from '/microsoft.svg';
import { useAuth } from '../../auth/useAuth';

const SocialLoginButtons: React.FC = () => {
  const { instance } = useMsal();
  const { setAuth } = useAuth();

  const handleAzureLogin = () => {
    instance
      .loginPopup({
        scopes: ['user.read'],
      })
      .then(response => {
        const token = response.uniqueId;
        if (token) {
          setAuth(token);
          console.log('Autenticado com sucesso', response);
        }
      })
      .catch(error => {
        console.error('Erro de autenticação:', error);
      });
  };
  return (
    <div className='flex flex-col items-center space-y-4'>
      <button
        className='flex items-center w-1/2 justify-center rounded-md bg-white border border-black/20 px-3 py-1.5 text-sm font-semibold leading-6 text-black shadow-md hover:bg-gray-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={handleAzureLogin}
      >
        <img src={microsoft} alt='microsoft icon' className='mr-4 h-5 w-5' />
        Microsoft
      </button>

      {/* <button
        className='flex w-full justify-center rounded-md bg-gray-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-gray-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={() => alert('Continue with GitHub')}
      >
        Login with GitHub
      </button> */}
    </div>
  );
};

export default SocialLoginButtons;
