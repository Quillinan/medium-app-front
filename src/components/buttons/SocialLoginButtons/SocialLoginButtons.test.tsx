import { render, screen, fireEvent } from '@testing-library/react';
import SocialLoginButtons from './SocialLoginButtons';
import { useMsal } from '@azure/msal-react';
import { useAuth } from '@auth/useAuth';
import { Mock } from 'vitest';

vi.mock('@azure/msal-react', () => ({
  useMsal: vi.fn(),
}));

vi.mock('@auth/useAuth', () => ({
  useAuth: vi.fn(),
}));

describe('SocialLoginButtons', () => {
  const mockLoginPopup = vi.fn();
  const mockSetAuth = vi.fn();

  beforeAll(() => {
    vi.spyOn(console, 'error').mockImplementation(() => {});
  });

  afterAll(() => {
    vi.restoreAllMocks();
  });

  beforeEach(() => {
    vi.clearAllMocks();

    (useMsal as Mock).mockReturnValue({
      instance: {
        loginPopup: mockLoginPopup,
      },
    });

    (useAuth as Mock).mockReturnValue({
      setAuth: mockSetAuth,
    });
  });

  it('should handle login errors', async () => {
    const error = new Error('Erro de autenticação');
    mockLoginPopup.mockRejectedValueOnce(error);

    render(<SocialLoginButtons />);

    const microsoftButton = screen.getByRole('button', { name: /Microsoft/i });
    fireEvent.click(microsoftButton);

    expect(mockLoginPopup).toHaveBeenCalled();

    expect(mockSetAuth).not.toHaveBeenCalled();
  });

  it('should render the Microsoft login button', () => {
    render(<SocialLoginButtons />);

    const microsoftButton = screen.getByRole('button', { name: /Microsoft/i });
    expect(microsoftButton).toBeInTheDocument();
  });

  it('should call loginPopup and setAuth on successful login', async () => {
    mockLoginPopup.mockResolvedValueOnce({
      account: { name: 'mock-username' },
      uniqueId: 'mock-userId',
      accessToken: 'mock-token',
    });

    render(<SocialLoginButtons />);

    const microsoftButton = screen.getByRole('button', { name: /Microsoft/i });
    fireEvent.click(microsoftButton);

    expect(mockLoginPopup).toHaveBeenCalledWith({
      scopes: ['user.read'],
    });

    await screen.findByRole('button', { name: /Microsoft/i });

    expect(mockSetAuth).toHaveBeenCalledWith(
      'mock-token',
      'mock-userId',
      'mock-username'
    );
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<SocialLoginButtons />);

    expect(asFragment()).toMatchSnapshot();
  });
});
