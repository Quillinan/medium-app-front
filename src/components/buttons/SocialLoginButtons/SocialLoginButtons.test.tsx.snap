// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialLoginButtons > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center space-y-4"
  >
    <button
      class="flex items-center w-1/2 justify-center rounded-md bg-white border border-black/20 px-3 py-1.5 text-sm font-semibold leading-6 text-black shadow-md hover:bg-gray-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600"
    >
      <img
        alt="microsoft icon"
        class="mr-4 h-5 w-5"
        src="/microsoft.svg"
      />
      Microsoft
    </button>
  </div>
</DocumentFragment>
`;
